### PR TITLE
Split `InvokeContext` into two

### PR DESCRIPTION
**Pull request by mriise**
Splits `InvokeContext` into `NetworkContext` and `MessageContext` for clearer separation of capability domains (network context must be able to inspect network values, same with message/message context).

`vm` module is where i kept the message context as it doesnt really fit anywhere else and I didnt want to have it in it's own module.

!!! a test was disabled till I can merge a fix into builtin actors for dependencies of `vm::context` !!!

### CODE REVIEW

File: fvm/src/syscalls/network.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 use anyhow::Context as _;
-use fvm_shared::sys;
+use fvm_shared::sys::out::network::SyscallNetworkContext;
+use fvm_shared::sys::{self};
```
Comment (Stebalien) [inline fvm/src/syscalls/network.rs:3]: ?

File: shared/src/sys/out.rs [hunk 2]
```diff
@@ -57,30 +57,37 @@ pub mod crypto {
...
-    use crate::clock::ChainEpoch;
     use crate::sys::TokenAmount;
     use crate::{ActorID, MethodNum};
 
     #[derive(Debug, Copy, Clone)]
     #[repr(packed, C)]
-    pub struct InvocationContext {
-        /// The value that was received.
-        pub value_received: TokenAmount,
+    pub struct SyscallMessageContext {
```
Comment (Stebalien) [inline shared/src/sys/out.rs:65]:
```suggestion
    pub struct MessageContext {
```

?

File: testing/integration/tests/address_test.rs [hunk 3]
```diff
@@ -11,8 +11,9 @@ use fvm_shared::state::StateTreeVersion;
 use fvm_shared::version::NetworkVersion;
 use num_traits::Zero;
 
-#[test]
-fn basic_address_tests() {
+// XXX RE-ENABLE THIS AFTER MERGE WITH BUILTIN ACTORS
+// #[test]
+fn _basic_address_tests() {
```
Comment (Stebalien) [inline testing/integration/tests/address_test.rs:16]: `#[cfg(ignore)]`?

File: sdk/src/vm.rs [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
+use fvm_shared::sys::out::vm::SyscallMessageContext;
 
 use crate::sys;
 
 /// BlockID representing nil parameters or return data.
 pub const NO_DATA_BLOCK_ID: u32 = 0;
 
 lazy_static::lazy_static! {
-    pub(crate) static ref INVOCATION_CONTEXT: InvocationContext = {
+    pub(crate) static ref MESSAGE_CONTEXT: SyscallMessageContext = {
```
Comment (Stebalien) [inline sdk/src/vm.rs:12]: move to `message`.

**Review by Stebalien — COMMENTED** [review]
mostly nits

File: shared/src/sys/out.rs (hunk 2 above)
Comment (mriise) [reply shared/src/sys/out.rs:65]: I would like to keep it consistent with `SyscallNetworkContext`

File: testing/integration/tests/address_test.rs (hunk 3 above)
Comment (mriise) [reply testing/integration/tests/address_test.rs:16]: then it makes the imports all mad so... eh

File: shared/src/sys/out.rs (hunk 2 above)
Comment (Stebalien) [reply shared/src/sys/out.rs:65]: I mean, rename both.

File: testing/integration/tests/address_test.rs (hunk 3 above)
Comment (Stebalien) [reply testing/integration/tests/address_test.rs:16]: Ah, ok, fine.

**Review by Stebalien — APPROVED** [review]

**Pull request merged**